File: backend/src/admin/objects/StoreOnboardingRequest.scala
package delivery.admin.objects

final case class StoreOnboardingRequest(
    id: String,
    ownerUsername: String,
    storeName: String,
    address: String,
    description: String,
    status: String,
    rejectionReason: Option[String],
    reviewedBy: Option[String],
    createdAt: String,
    reviewedAt: Option[String]
)

File: frontend/src/objects/admin/StoreOnboardingRequest.ts
export type StoreOnboardingStatus = 'pending' | 'accepted' | 'rejected'

export interface StoreOnboardingRequest {
  id: string
  ownerUsername: string
  storeName: string
  address: string
  description: string
  tags: string[]
  status: StoreOnboardingStatus
  rejectionReason?: string | null
  reviewedBy?: string | null
  createdAt: string
  reviewedAt?: string | null
}
